code/datums/components/riding.dm

```diff
@@ -47,6 +48,11 @@
 /datum/component/riding/proc/set_vehicle_dir_layer(dir, layer)
 	directional_vehicle_layers["[dir]"] = layer
 
+/datum/component/riding/proc/vehicle_dir_changed(from_dir, to_dir)
+	spawn
```

Comment:
Wait, what the hell?

Reply:
basically the signal fires *before* the dir is actually changed so uh this is necessary

Reply:
Have you considered adding a "dir override" argument to both procs below at least?

Reply:
addtimer's better tbh